src/libraries/System.Net.Sockets/src/System/Net/Sockets/Socket.cs

```diff
@@ -1161,6 +1161,11 @@ public int Send(IList<ArraySegment<byte>> buffers, SocketFlags socketFlags, out
 
             int bytesTransferred;
             errorCode = SocketPal.Send(_handle, buffers, socketFlags, out bytesTransferred);
+            if (SocketsTelemetry.Log.IsEnabled())
+            {
+                SocketsTelemetry.Log.BytesSent(bytesTransferred);
+                if (SocketType == SocketType.Dgram) SocketsTelemetry.Log.DatagramSent();
```

Comment:
The logging currently treats datagrams as being sent even if errorCode != success. I'd guess customers will want the counter to instead measure the number of datagrams that were actually sent, not the number that we attempted to send.
Similar principle for bytesTransfered but I assume in the error case bytesTransfered = 0 and we are getting that behavior for free.

Reply:
Good point. Fixed.